### ISSUE TITLE
Node.js version specific Babel presets

### ISSUE THREAD

**Issue by sindresorhus** [avajs/ava#148]
So Babel only transpiles features not already supported by the platform.

@jamestalmage said he's interested in looking into this.

**Comment by jamestalmage** [comment]
Looks like work has already begun:

https://www.npmjs.com/package/babel-preset-es2015-node5
https://www.npmjs.com/package/babel-preset-es2015-node4
https://www.npmjs.com/package/babel-preset-node5

Still, I think it might be better to automate this.

I am thinking of create a test-file for each of the [available plugins](https://babeljs.io/docs/plugins/#es2015) that uses the specific feature that plugin provides.

Then a script to require each in a try/catch and see if they throw / fail to compile. That gives us a nice automated update path as future versions are released. I am hoping babel's own test suite will make this all a simple matter of cutting and pasting code samples... There are a lot of plugins.

**Comment by sindresorhus** [comment]
Wouldn't live feature testing be slow? I had a similar idea of doing the feature testing for every major Node.js version before publishing.

**Comment by jamestalmage** [comment]
Yes - as part of the update process. Not live testing.

After the test suite runs we output a file for that specific version in a subfolder:

```js
// index.js
module.exports = require('./plugin-lists/' + process.version);
```

or something like that.

**Comment by sindresorhus** [comment]
:+1: That's exactly what I was thinking.

**Comment by jamestalmage** [comment]
It might be cool to fetch the actual plugins from `npm` on demand as well (so `npm install` goes faster in newer versions of node). We could give back a few CPU cycles to travis-ci.org that way.

Definitely not something to target for the first release, but might be interesting to play with down the road.

**Comment by hzoo** [comment]
Haven't implemented it yet but I would like to see at least a preset where you can specify that environment you want to target and it would have a predetermined list of transforms based on what it supports. You could just use the data or use `env` to do different things. https://github.com/babel/babel/pull/3476

**Comment by novemberborn** [comment]
Just saw this: https://github.com/gabmontes/babel-preset-latest-minimal

**Comment by sindresorhus** [comment]
@novemberborn I'm a little bit sceptical of feature testing. New features might be buggy or slow. I would be more comfortable with a preset that choose plugins based on vetted testing on various Node.js versions. Maybe https://github.com/babel/babel-preset-env when it gets more mature.

**Comment by novemberborn** [comment]
Is fixed by #1193.
